mcp/src/main/java/io/modelcontextprotocol/spec/AsyncCloseable.java

```diff
@@ -0,0 +1,24 @@
+package io.modelcontextprotocol.spec;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * Interface for close operations that are asynchronous.
+ */
+public interface AsyncCloseable {
+
+	/**
+	 * Begins the process of closing the resource gracefully, if there is one in progress,
+	 * return the existing one.
```

Comment:
I don't think that's accurate? We'd need to keep track of closing-in-progress and I am not certain we do that currently.

Reply:
So yo suggest we drop the :
```
if there is one in progress, return the existing one.
```
?

I think we should always return the same instance.

Reply:
I checked the code, you are right, not all implementation return the same `Mono`, will drop this description.